### PR TITLE
API - business allowable actions for comp auths

### PR DESCRIPTION
**Pull request by kialj876**
*Issue #:* /bcgov/entity#22030

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/services/authz.py [hunk 1]
```diff
@@ -82,33 +116,9 @@ def authorized(  # pylint: disable=too-many-return-statements
         if any(elem in action for elem in staff_only_actions):
             return False
 
-        template_url = current_app.config.get('AUTH_SVC_URL')
-        auth_url = f'{template_url}/entities/{identifier}/authorizations'
-
-        token = jwt.get_token_auth_header()
-        headers = {'Authorization': 'Bearer ' + token}
-        try:
```
Comment (kialj876) [inline legal-api/src/legal_api/services/authz.py:90]: moved this into `_call_auth_api` so that it could be reused in `get_account_products`

**Comment by kialj876** [comment]
tests are passing, but codecov is erring due to external issue I think

**Review by JazzarKarim — APPROVED** [review]
LGTM 👍

**Comment by JazzarKarim** [comment]
> tests are passing, but codecov is erring due to external issue I think

The codecov issue has been there for a while. It's there for both FE and BE. We fixed it a while back but now the issue is back. No worries, we can ignore it for now.

File: legal-api/src/legal_api/services/authz.py [hunk 2]
```diff
@@ -71,9 +104,10 @@ def authorized(  # pylint: disable=too-many-return-statements
             or jwt.validate_roles([COLIN_SVC_ROLE]):
         return True
 
-    # allow IDIM view access on everything
-    if len(action) == 1 and action[0] == 'view' and jwt.validate_roles([ACCOUNT_IDENTITY]):
-        return True
+    # allow IDIM and Competent Authorities view access on everything
+    if len(action) == 1 and action[0] == 'view':
```
Comment (hfekete) [inline legal-api/src/legal_api/services/authz.py:108]: Just a note
You could have put 2nd if expression in the bracket and join it with `and`, if there will be no granularity needed for some other options.

**Review by hfekete — APPROVED** [review]
Looks good from my perspective, lets see what comments others have.

**Review by argush3 — APPROVED** [review]

**Review by chenhongjing — APPROVED** [review]

File: legal-api/src/legal_api/services/authz.py (hunk 2 above)
Comment (kialj876) [reply legal-api/src/legal_api/services/authz.py:108]: yeah I just separated them for readability

**Pull request merged**